frontend/src/registerSW.ts

```diff
@@ -0,0 +1,27 @@
+if ('serviceWorker' in navigator) {
+  window.addEventListener('load', () => {
+    navigator.serviceWorker.register('/sw.js', { scope: '/' }).then(() => {
+      console.log('SW registered')
+    })
+  })
+}
+
+if ('serviceWorker' in navigator) {
+  navigator.serviceWorker.register('/sw.js').then((registration) => {
+    registration.onupdatefound = () => {
+      const newWorker = registration.installing
+      if (newWorker) {
+        newWorker.onstatechange = () => {
+          if (newWorker.state === 'installed') {
+            // Prüfen, ob ein Update vorhanden ist
+            if (navigator.serviceWorker.controller) {
+              // Neuer Inhalt verfügbar, Nutzer benachrichtigen
+              const updateBanner = document.getElementById('update-banner')
```

Comment:
Wieso benachritigen wir den Nutzer? Können wir das update nicht einfach laden ohne zu fragen?

Reply:
Möglich ist das, braucht ein wenig Zeit. Habe es nicht gemacht, weil in der Theorie Nutzerformulareingaben verloren gehen könnten, weil sich die Seite ohne seine Zustimmung aktualisiert.In der Praxis ist es unrealistisch, dass er in 2-3 Sekunden ein Formular ausgefüllt hat denke ich.

Reply:
Mir ist aufgefallen, bei der Umsetzung mit dem automatischen Reload, dass wenn im Hintergrund ein weiterer Tab geöffnet ist, bei dem vielleicht ein Formular mit Eingaben offen ist, dieses automatisch auch aktualisiert wird. 
Das wiederum wäre nicht sonderlich toll, da hier dann Daten verloren gehen könnten.